src/contract/__test__/contract-interact.ts

```diff
@@ -0,0 +1,152 @@
+    contract: "io186s45j3rgvhxh25ec6xk9wap0drtthk3jq4du7",
+    amount: "0",
+    data: Buffer.from(encodeInputData(abiFunctions, "set", { x: 10 }), "hex")
+  });
+  const reps = await method.execute();
+  t.truthy(reps);
+
+  await sleepPromise(30000);
+
+  const actions = await client.getActions({
+    byHash: { actionHash: reps, checkingPending: true }
+  });
+  t.deepEqual(actions.actionInfo.length, 1, "contract action is empty");
+  const result = await client.readContract({
+    action: actions.actionInfo[0].action
+  });
+  t.truthy(result);
+  t.deepEqual(result.data, "");
+});
+
+test.skip("Contract_get_SimpleStorage", async t => {
+  const solFile = "./SimpleStorage.sol";
+  const contractName = ":SimpleStorage";
+  const input = fs.readFileSync(path.resolve(__dirname, solFile));
+  const output = solc.compile(input.toString(), 1);
+  const contract = output.contracts[contractName];
+  const abi = JSON.parse(contract.interface);
+  const abiFunctions = getAbiFunctions(abi);
+
+  const client = new RpcMethod(TEST_HOSTNAME, { timeout: 10000 });
+  const sender = Account.fromPrivateKey(TEST_ACCOUNT.privateKey);
+  const method = new ExecutionMethod(client, sender, {
+    gasPrice: "1000000000000",
+    gasLimit: "1000000",
+    contract: "io186s45j3rgvhxh25ec6xk9wap0drtthk3jq4du7",
+    amount: "0",
+    data: Buffer.from(encodeInputData(abiFunctions, "get", {}), "hex")
+  });
+  const action = await method.sign();
+
+  const result = await client.readContract({ action: action });
+  t.truthy(result);
+});
+
+test.skip("Contract_method_get_SimpleStorage", async t => {
+  const solFile = "./SimpleStorage.sol";
+  const contractName = ":SimpleStorage";
+  const input = fs.readFileSync(path.resolve(__dirname, solFile));
+  const output = solc.compile(input.toString(), 1);
+  const contractDef = output.contracts[contractName];
+
+  const contract = new Contract(
+    JSON.parse(contractDef.interface),
+    "io186s45j3rgvhxh25ec6xk9wap0drtthk3jq4du7"
+  );
+
+  const client = new RpcMethod(TEST_HOSTNAME, { timeout: 10000 });
+  const sender = Account.fromPrivateKey(TEST_ACCOUNT.privateKey);
+  const result = await contract.methods.get({
+    client: client,
+    account: sender,
+    gasLimit: "1000000",
+    gasPrice: "1000000000000"
+  });
+  t.truthy(result);
+});
+
+test.skip("Contract_method_set_SimpleStorage", async t => {
+  const solFile = "./SimpleStorage.sol";
+  const contractName = ":SimpleStorage";
+  const input = fs.readFileSync(path.resolve(__dirname, solFile));
+  const output = solc.compile(input.toString(), 1);
+  const contractDef = output.contracts[contractName];
+
+  const contract = new Contract(
+    JSON.parse(contractDef.interface),
+    "io186s45j3rgvhxh25ec6xk9wap0drtthk3jq4du7"
+  );
+
+  const client = new RpcMethod(TEST_HOSTNAME, { timeout: 10000 });
+  const sender = Account.fromPrivateKey(TEST_ACCOUNT.privateKey);
+  const result = await contract.methods.set(
+    {
+      client: client,
```

Comment:
1. let's not call them client, call them provider
2. accept provider in the constructor instead of here